# 6.1: Sourcing Open Data

## Table of Content: 

### Task 1 - Source your data. Use the requirements (and your own interests) to source an open data set from the web. Import the Data Set

### Task 2 - Clean your data. Conduct some basic data cleaning and consistency checks in Jupyter to ensure your data is ready for further analysis.

### Task 3 - Understand your data. Develop a basic understanding of your data set by reviewing the variables and performing basic descriptive statistical analysis.

***
***

## Task 1 - Import Libraries and Data Set

In [1]:
import pandas as pd
import numpy as np
import os

### Import Files

In [2]:
gun_violence_data = pd.read_csv(r'C:\Users\facun\Desktop\Data Analysis\CF\Data Immersion\Achievement 6\Ex 6.1\Gun Violence Data\gun-violence-data_01-2013_03-2018.csv', index_col = False)

## Task 2 - Clean the Data Set.

### EDA

In [3]:
gun_violence_data

incident_id        date           state city_or_county  \
0            461105  2013-01-01    Pennsylvania     Mckeesport   
1            460726  2013-01-01      California      Hawthorne   
2            478855  2013-01-01            Ohio         Lorain   
3            478925  2013-01-05        Colorado         Aurora   
4            478959  2013-01-07  North Carolina     Greensboro   
...             ...         ...             ...            ...   
239672      1083142  2018-03-31       Louisiana          Rayne   
239673      1083139  2018-03-31       Louisiana   Natchitoches   
239674      1083151  2018-03-31       Louisiana         Gretna   
239675      1082514  2018-03-31           Texas        Houston   
239676      1081940  2018-03-31           Maine   Norridgewock   

                                          address  n_killed  n_injured  \
0       1506 Versailles Avenue and Coursin Street         0          4   
1                    13500 block of Cerise Avenue         1          3   
2                           1776 East 28th Street         1          3   
3                16000 block of East Ithaca Place         4          0   
4                       307 Mourning Dove Terrace         2          2   
...                                           ...       ...        ...   
239672         North Riceland Road and Highway 90         0          0   
239673                             247 Keyser Ave         1          0   
239674                  1300 block of Cook Street         0          1   
239675                     12630 Ashford Point Dr         1          0   
239676                           434 Skowhegan Rd         2          0   

                                             incident_url  \
0       http://www.gunviolencearchive.org/incident/461105   
1       http://www.gunviolencearchive.org/incident/460726   
2       http://www.gunviolencearchive.org/incident/478855   
3       http://www.gunviolencearchive.org/incident/478925   
4       http://www.gunviolencearchive.org/incident/478959   
...                                                   ...   
239672  http://www.gunviolencearchive.org/incident/108...   
239673  http://www.gunviolencearchive.org/incident/108...   
239674  http://www.gunviolencearchive.org/incident/108...   
239675  http://www.gunviolencearchive.org/incident/108...   
239676  http://www.gunviolencearchive.org/incident/108...   

                                               source_url  \
0       http://www.post-gazette.com/local/south/2013/0...   
1       http://www.dailybulletin.com/article/zz/201301...   
2       http://chronicle.northcoastnow.com/2013/02/14/...   
3       http://www.dailydemocrat.com/20130106/aurora-s...   
4       http://www.journalnow.com/news/local/article_d...   
...                                                   ...   
239672  http://www.klfy.com/news/local/rayne-woman-cha...   
239673  http://www.ksla.com/story/37854648/man-wanted-...   
239674  http://www.nola.com/crime/index.ssf/2018/04/sh...   
239675  https://www.chron.com/news/houston-texas/houst...   
239676  https://www.centralmaine.com/2018/03/31/police...   

        incident_url_fields_missing  ...                    participant_age  \
0                             False  ...                              0::20   
1                             False  ...                              0::20   
2                             False  ...  0::25||1::31||2::33||3::34||4::33   
3                             False  ...         0::29||1::33||2::56||3::33   
4                             False  ...         0::18||1::46||2::14||3::47   
...                             ...  ...                                ...   
239672                        False  ...                              0::25   
239673                        False  ...                              1::21   
239674                        False  ...                              0::21   
239675                        False  ...                              

In [4]:
gun_violence_data.head()

incident_id        date           state city_or_county  \
0       461105  2013-01-01    Pennsylvania     Mckeesport   
1       460726  2013-01-01      California      Hawthorne   
2       478855  2013-01-01            Ohio         Lorain   
3       478925  2013-01-05        Colorado         Aurora   
4       478959  2013-01-07  North Carolina     Greensboro   

                                     address  n_killed  n_injured  \
0  1506 Versailles Avenue and Coursin Street         0          4   
1               13500 block of Cerise Avenue         1          3   
2                      1776 East 28th Street         1          3   
3           16000 block of East Ithaca Place         4          0   
4                  307 Mourning Dove Terrace         2          2   

                                        incident_url  \
0  http://www.gunviolencearchive.org/incident/461105   
1  http://www.gunviolencearchive.org/incident/460726   
2  http://www.gunviolencearchive.org/incident/478855   
3  http://www.gunviolencearchive.org/incident/478925   
4  http://www.gunviolencearchive.org/incident/478959   

                                          source_url  \
0  http://www.post-gazette.com/local/south/2013/0...   
1  http://www.dailybulletin.com/article/zz/201301...   
2  http://chronicle.northcoastnow.com/2013/02/14/...   
3  http://www.dailydemocrat.com/20130106/aurora-s...   
4  http://www.journalnow.com/news/local/article_d...   

   incident_url_fields_missing  ...                    participant_age  \
0                        False  ...                              0::20   
1                        False  ...                              0::20   
2                        False  ...  0::25||1::31||2::33||3::34||4::33   
3                        False  ...         0::29||1::33||2::56||3::33   
4                        False  ...         0::18||1::46||2::14||3::47   

                               participant_age_group  \
0  0::Adult 18+||1::Adult 18+||2::Adult 18+||3::A...   
1  0::Adult 18+||1::Adult 18+||2::Adult 18+||3::A...   
2  0::Adult 18+||1::Adult 18+||2::Adult 18+||3::A...   
3  0::Adult 18+||1::Adult 18+||2::Adult 18+||3::A...   
4  0::Adult 18+||1::Adult 18+||2::Teen 12-17||3::...   

                            participant_gender  \
0         0::Male||1::Male||3::Male||4::Female   
1                                      0::Male   
2  0::Male||1::Male||2::Male||3::Male||4::Male   
3         0::Female||1::Male||2::Male||3::Male   
4       0::Female||1::Male||2::Male||3::Female   

                                    participant_name  \
0                                     0::Julian Sims   
1                                  0::Bernard Gillis   
2  0::Damien Bell||1::Desmen Noble||2::Herman Sea...   
3  0::Stacie Philbrook||1::Christopher Ratliffe||...   
4  0::Danielle Imani Jameison||1::Maurice Eugene ...   

   participant_relationship  \
0                       NaN   
1                       NaN   
2                       NaN   
3                       NaN   
4                 3::Family   

                                  participant_status  \
0  0::Arrested||1::Injured||2::Injured||3::Injure...   
1      0::Killed||1::Injured||2::Injured||3::Injured   
2  0::Injured, Unharmed, Arrested||1::Unharmed, A...   
3         0::Killed||1::Killed||2::Killed||3::Killed   
4       0::Injured||1::Injured||2::Killed||3::Killed   

                                    participant_type  \
0  0::Victim||1::Victim||2::Victim||3::Victim||4:...   
1  0::Victim||1::Victim||2::Victim||3::Victim||4:...   
2  0::Subject-Suspect||1::Subject-Suspect||2::Vic...   
3  0::Victim||1::Victim||2::Victim||3::Subject-Su...   
4  0::Victim||1::Victim||2::Victim||3::Subject-Su...   

                                             sources state_house_district  \
0  http://pittsburgh.cbslocal.com/2013/01/01/4-pe...                  NaN   
1  http://losangeles.cbslocal.com/2013/01/01/man-...                 62.0   
2  http://www.morningjournal.com/ge

In [5]:
gun_violence_data.tail()

incident_id        date      state city_or_county  \
239672      1083142  2018-03-31  Louisiana          Rayne   
239673      1083139  2018-03-31  Louisiana   Natchitoches   
239674      1083151  2018-03-31  Louisiana         Gretna   
239675      1082514  2018-03-31      Texas        Houston   
239676      1081940  2018-03-31      Maine   Norridgewock   

                                   address  n_killed  n_injured  \
239672  North Riceland Road and Highway 90         0          0   
239673                      247 Keyser Ave         1          0   
239674           1300 block of Cook Street         0          1   
239675              12630 Ashford Point Dr         1          0   
239676                    434 Skowhegan Rd         2          0   

                                             incident_url  \
239672  http://www.gunviolencearchive.org/incident/108...   
239673  http://www.gunviolencearchive.org/incident/108...   
239674  http://www.gunviolencearchive.org/incident/108...   
239675  http://www.gunviolencearchive.org/incident/108...   
239676  http://www.gunviolencearchive.org/incident/108...   

                                               source_url  \
239672  http://www.klfy.com/news/local/rayne-woman-cha...   
239673  http://www.ksla.com/story/37854648/man-wanted-...   
239674  http://www.nola.com/crime/index.ssf/2018/04/sh...   
239675  https://www.chron.com/news/houston-texas/houst...   
239676  https://www.centralmaine.com/2018/03/31/police...   

        incident_url_fields_missing  ...  participant_age  \
239672                        False  ...            0::25   
239673                        False  ...            1::21   
239674                        False  ...            0::21   
239675                        False  ...            0::42   
239676                        False  ...     0::58||1::62   

             participant_age_group  participant_gender  \
239672                0::Adult 18+           0::Female   
239673  0::Adult 18+||1::Adult 18+    0::Male||1::Male   
239674                0::Adult 18+             0::Male   
239675                0::Adult 18+             0::Male   
239676  0::Adult 18+||1::Adult 18+  0::Female||1::Male   

                                      participant_name  \
239672                                0::Jhkeya Tezeno   
239673  0::Jamal Haskett||1::Jaquarious Tyjuan Ardison   
239674                                             NaN   
239675                                  0::Leroy Ellis   
239676        0::Marie Lancaster Hale||1::William Hale   

                         participant_relationship  \
239672                                        NaN   
239673                                        NaN   
239674                                        NaN   
239675                                        NaN   
239676  1::Significant others - current or former   

                      participant_status               participant_type  \
239672             0::Unharmed, Arrested             0::Subject-Suspect   
239673  0::Killed||1::Unharmed, Arrested  0::Victim||1::Subject-Suspect   
239674                        0::Injured                      0::Victim   
239675                         0::Killed                      0::Victim   
239676              0::Killed||1::Killed  0::Victim||1::Subject-Suspect   

                                                  sources  \
239672  http://www.klfy.com/news/local/rayne-woman-cha...   
239673  http://www.ksla.com/story/37854648/man-wanted-...   
239674  http://www.nola.com/crime/index.ssf/2018/04/sh...   
239675  http://www.khou.com/article/news/hpd-investiga...   
239676  https://www.centralmaine.com/2018/03/31/police...   

       state_house_district state_senate_district  
239672                  NaN                   NaN  
239673                 23.0                  31.0  
239674                 85.0                   7.0  
239675                149.0                  17.0  
239676                111.0               

In [6]:
gun_violence_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 239677 entries, 0 to 239676
Data columns (total 29 columns):
 #   Column                       Non-Null Count   Dtype  
---  ------                       --------------   -----  
 0   incident_id                  239677 non-null  int64  
 1   date                         239677 non-null  object 
 2   state                        239677 non-null  object 
 3   city_or_county               239677 non-null  object 
 4   address                      223180 non-null  object 
 5   n_killed                     239677 non-null  int64  
 6   n_injured                    239677 non-null  int64  
 7   incident_url                 239677 non-null  object 
 8   source_url                   239209 non-null  object 
 9   incident_url_fields_missing  239677 non-null  bool   
 10  congressional_district       227733 non-null  float64
 11  gun_stolen                   140179 non-null  object 
 12  gun_type                     140226 non-null  object 
 13 

In [7]:
gun_violence_data.describe().T

count           mean            std         min  \
incident_id             239677.0  559334.346404  293128.684285  92114.0000   
n_killed                239677.0       0.252290       0.521779      0.0000   
n_injured               239677.0       0.494007       0.729952      0.0000   
congressional_district  227733.0       8.001265       8.480835      0.0000   
latitude                231754.0      37.546598       5.130763     19.1114   
longitude               231754.0     -89.338348      14.359546   -171.4290   
n_guns_involved         140226.0       1.372442       4.678202      1.0000   
state_house_district    200905.0      55.447132      42.048117      1.0000   
state_senate_district   207342.0      20.477110      14.204560      1.0000   

                                  25%          50%            75%  \
incident_id             308545.000000  543587.0000  817228.000000   
n_killed                     0.000000       0.0000       0.000000   
n_injured                    0.000000       0.0000       1.000000   
congressional_district       2.000000       5.0000      10.000000   
latitude                    33.903400      38.5706      41.437375   
longitude                  -94.158725     -86.2496     -80.048625   
n_guns_involved              1.000000       1.0000       1.000000   
state_house_district        21.000000      47.0000      84.000000   
state_senate_district        9.000000      19.0000      30.000000   

                                 max  
incident_id             1.083472e+06  
n_killed                5.000000e+01  
n_injured               5.300000e+01  
congressional_district  5.300000e+01  
latitude                7.133680e+01  
longitude               9.743310e+01  
n_guns_involved         4.000000e+02  
state_house_district    9.010000e+02  
state_senate_district   9.400000e+01

In [8]:
# Checking for missing values in each column

missing_values = gun_violence_data.isnull().sum()
missing_values

incident_id                         0
date                                0
state                               0
city_or_county                      0
address                         16497
n_killed                            0
n_injured                           0
incident_url                        0
source_url                        468
incident_url_fields_missing         0
congressional_district          11944
gun_stolen                      99498
gun_type                        99451
incident_characteristics          326
latitude                         7923
location_description           197588
longitude                        7923
n_guns_involved                 99451
notes                           81017
participant_age                 92298
participant_age_group           42119
participant_gender              36362
participant_name               122253
participant_relationship       223903
participant_status              27626
participant_type                24863
sources     

In [9]:
# Calculating the percentage of missing values

percentage_missing = (missing_values / len(gun_violence_data)) * 100

# Combining the missing values count and percentage into a DataFrame

missing_data_analysis = pd.DataFrame({

    'Missing Values': missing_values, 

    'Percentage': percentage_missing

})



missing_data_analysis.sort_values(by='Percentage', ascending=False)

Missing Values  Percentage
participant_relationship             223903   93.418643
location_description                 197588   82.439283
participant_name                     122253   51.007397
gun_stolen                            99498   41.513370
n_guns_involved                       99451   41.493760
gun_type                              99451   41.493760
participant_age                       92298   38.509327
notes                                 81017   33.802576
participant_age_group                 42119   17.573234
state_house_district                  38772   16.176771
participant_gender                    36362   15.171251
state_senate_district                 32335   13.491073
participant_status                    27626   11.526346
participant_type                      24863   10.373544
address                               16497    6.883013
congressional_district                11944    4.983373
latitude                               7923    3.305699
longitude                              7923    3.305699
sources                                 609    0.254092
source_url                              468    0.195263
incident_characteristics                326    0.136016
date                                      0    0.000000
incident_url_fields_missing               0    0.000000
incident_url                              0    0.000000
n_injured                                 0    0.000000
n_killed                                  0    0.000000
city_or_county                            0    0.000000
state                                     0    0.000000
incident_id                               0    0.000000

#### Here's a summary of the key findings:

1.  participant_relationship: 93.42% missing.
2.  location_description: 82.44% missing.
3.  participant_name: 51.01% missing.
4.  gun_stolen: 41.51% missing.
5.  n_guns_involved: 41.49% missing.
6.  gun_type: 41.49% missing.
7.  participant_age: 38.51% missing.
8.  notes: 33.80% missing.
9.  participant_age_group: 17.57% missing.
10. state_house_district: 16.18% missing.
11. participant_gender: 15.17% missing.
12. state_senate_district: 13.49% missing.
13. participant_status: 11.53% missing.
14. participant_type: 10.37% missing.
15. address: 6.88% missing.
16. congressional_district: 4.98% missing.
17. latitude: 3.31% missing.
18. longitude: 3.31% missing.
19. sources: 0.25% missing.
20. source_url: 0.20% missing.
21. incident_characteristics: 0.14% missing.

The rest of the columns, including 'date', 'incident_url_fields_missing', 'incident_url', 'n_injured', 'n_killed', 'city_or_county', 'state', and 'incident_id', have no missing values.

In [10]:
gun_violence_data.shape

(239677, 29)

### Data Cleaning

In [11]:
# Removing columns with high percentage of missing values and participant_name for privacy concerns
columns_to_remove = ['participant_relationship', 'location_description', 'participant_name']
cleaned_gun_violence_data = gun_violence_data.drop(columns=columns_to_remove)

In [12]:
# Checking the shape of the dataset after removal
cleaned_dataset_shape = cleaned_gun_violence_data.shape

cleaned_dataset_shape

(239677, 26)

In [13]:
# Further removing 'sources' and 'source_url' columns as they are deemed irrelevant for the analysis
additional_columns_to_remove = ['sources', 'source_url']
cleaned_gun_violence_data = cleaned_gun_violence_data.drop(columns=additional_columns_to_remove)

# Checking the shape of the dataset after the additional removal
cleaned_gun_violence_data.shape


(239677, 24)

In [14]:
# Removing the 'incident_url_fields_missing' and 'incident_url' columns

columns_to_remove = ['incident_url_fields_missing', 'incident_url']

cleaned_gun_violence_data = cleaned_gun_violence_data.drop(columns=columns_to_remove)



# Checking the shape of the dataset after the removal

cleaned_gun_violence_data.shape


(239677, 22)

In [15]:
# Removing the 'notes'

columns_to_remove = ['notes']

cleaned_gun_violence_data = cleaned_gun_violence_data.drop(columns=columns_to_remove)



# Checking the shape of the dataset after the removal

cleaned_gun_violence_data.shape

(239677, 21)

I created 2 new columns "victim_count"  and "suspect_count"

In [24]:
def count_victims_and_suspects(participant_type):
    if pd.isna(participant_type):
        return 0, 0  # Return 0, 0 if the data is missing

    victim_count = participant_type.count("Victim")
    suspect_count = participant_type.count("Suspect")
    
    return victim_count, suspect_count

# Apply the function to each row in the DataFrame
cleaned_gun_violence_data[['victim_count', 'suspect_count']] = cleaned_gun_violence_data['participant_type'].apply(lambda x: count_victims_and_suspects(x)).apply(pd.Series)





In [23]:
# Display few rows to verify the new columns

cleaned_gun_violence_data

incident_id        date           state city_or_county  \
0            461105  2013-01-01    Pennsylvania     Mckeesport   
1            460726  2013-01-01      California      Hawthorne   
2            478855  2013-01-01            Ohio         Lorain   
3            478925  2013-01-05        Colorado         Aurora   
4            478959  2013-01-07  North Carolina     Greensboro   
...             ...         ...             ...            ...   
239672      1083142  2018-03-31       Louisiana          Rayne   
239673      1083139  2018-03-31       Louisiana   Natchitoches   
239674      1083151  2018-03-31       Louisiana         Gretna   
239675      1082514  2018-03-31           Texas        Houston   
239676      1081940  2018-03-31           Maine   Norridgewock   

                                          address  n_killed  n_injured  \
0       1506 Versailles Avenue and Coursin Street         0          4   
1                    13500 block of Cerise Avenue         1          3   
2                           1776 East 28th Street         1          3   
3                16000 block of East Ithaca Place         4          0   
4                       307 Mourning Dove Terrace         2          2   
...                                           ...       ...        ...   
239672         North Riceland Road and Highway 90         0          0   
239673                             247 Keyser Ave         1          0   
239674                  1300 block of Cook Street         0          1   
239675                     12630 Ashford Point Dr         1          0   
239676                           434 Skowhegan Rd         2          0   

        congressional_district              gun_stolen  \
0                         14.0                 Unknown   
1                         43.0                 Unknown   
2                          9.0  0::Unknown||1::Unknown   
3                          6.0                 Unknown   
4                          6.0  0::Unknown||1::Unknown   
...                        ...                     ...   
239672                     NaN              0::Unknown   
239673                     4.0              0::Unknown   
239674                     2.0              0::Unknown   
239675                     9.0              0::Unknown   
239676                     2.0  0::Unknown||1::Unknown   

                      gun_type  ... n_guns_involved  \
0                      Unknown  ...         Unknown   
1                      Unknown  ...         Unknown   
2       0::Unknown||1::Unknown  ...             2.0   
3                      Unknown  ...         Unknown   
4       0::Handgun||1::Handgun  ...             2.0   
...                        ...  ...             ...   
239672              0::Unknown  ...             1.0   
239673              0::Unknown  ...             1.0   
239674              0::Unknown  ...             1.0   
239675              0::Unknown  ...             1.0   
239676  0::Handgun||1::Shotgun  ...             2.0   

                          participant_age  \
0                                   0::20   
1                                   0::20   
2       0::25||1::31||2::33||3::34||4::33   
3              0::29||1::33||2::56||3::33   
4              0::18||1::46||2::14||3::47   
...                                   ...   
239672                              0::25   
239673                              1::21   
239674                              0::21   
239675                              0::42   
239676                       0::58||1::62   

                                    participant_age_group  \
0       0::Adult 18+||1::Adult 18+||2::Adult 18+||3::A...   
1       0::Adult 18+||1::Adult 18+||2::Adult 18+||3::A...   
2       0::Adult 18+||1::Adult 18+||2::Adult 18+||3::A...   
3       0::Adult 18+||1::Adult 18+||2::Adult 18+||3::A...   
4       0::Adult 18+||1::Adult 18+||2::Teen 12-17||3::...   
...                                                   ...   


In [25]:
# Creating a new column 'participant_number'
cleaned_gun_violence_data['participant_number'] = cleaned_gun_violence_data['victim_count'] + cleaned_gun_violence_data['suspect_count']

In [26]:
cleaned_gun_violence_data

incident_id        date           state city_or_county  \
0            461105  2013-01-01    Pennsylvania     Mckeesport   
1            460726  2013-01-01      California      Hawthorne   
2            478855  2013-01-01            Ohio         Lorain   
3            478925  2013-01-05        Colorado         Aurora   
4            478959  2013-01-07  North Carolina     Greensboro   
...             ...         ...             ...            ...   
239672      1083142  2018-03-31       Louisiana          Rayne   
239673      1083139  2018-03-31       Louisiana   Natchitoches   
239674      1083151  2018-03-31       Louisiana         Gretna   
239675      1082514  2018-03-31           Texas        Houston   
239676      1081940  2018-03-31           Maine   Norridgewock   

                                          address  n_killed  n_injured  \
0       1506 Versailles Avenue and Coursin Street         0          4   
1                    13500 block of Cerise Avenue         1          3   
2                           1776 East 28th Street         1          3   
3                16000 block of East Ithaca Place         4          0   
4                       307 Mourning Dove Terrace         2          2   
...                                           ...       ...        ...   
239672         North Riceland Road and Highway 90         0          0   
239673                             247 Keyser Ave         1          0   
239674                  1300 block of Cook Street         0          1   
239675                     12630 Ashford Point Dr         1          0   
239676                           434 Skowhegan Rd         2          0   

        congressional_district              gun_stolen  \
0                         14.0                 Unknown   
1                         43.0                 Unknown   
2                          9.0  0::Unknown||1::Unknown   
3                          6.0                 Unknown   
4                          6.0  0::Unknown||1::Unknown   
...                        ...                     ...   
239672                     NaN              0::Unknown   
239673                     4.0              0::Unknown   
239674                     2.0              0::Unknown   
239675                     9.0              0::Unknown   
239676                     2.0  0::Unknown||1::Unknown   

                      gun_type  ...                    participant_age  \
0                      Unknown  ...                              0::20   
1                      Unknown  ...                              0::20   
2       0::Unknown||1::Unknown  ...  0::25||1::31||2::33||3::34||4::33   
3                      Unknown  ...         0::29||1::33||2::56||3::33   
4       0::Handgun||1::Handgun  ...         0::18||1::46||2::14||3::47   
...                        ...  ...                                ...   
239672              0::Unknown  ...                              0::25   
239673              0::Unknown  ...                              1::21   
239674              0::Unknown  ...                              0::21   
239675              0::Unknown  ...                              0::42   
239676  0::Handgun||1::Shotgun  ...                       0::58||1::62   

                                    participant_age_group  \
0       0::Adult 18+||1::Adult 18+||2::Adult 18+||3::A...   
1       0::Adult 18+||1::Adult 18+||2::Adult 18+||3::A...   
2       0::Adult 18+||1::Adult 18+||2::Adult 18+||3::A...   
3       0::Adult 18+||1::Adult 18+||2::Adult 18+||3::A...   
4       0::Adult 18+||1::Adult 18+||2::Teen 12-17||3::...   
...                                                   ...   
239672                                       0::Adult 18+   
239673                         0::Adult 18+||1::Adult 18+   
239674                                       0::Adult 18+   
239675                                       0::Adult 18+   
239676                         0::Adult 18+||1::Adult 18+   

       

# Saving and Export

In [27]:
cleaned_gun_violence_data.to_csv(r'C:\Users\facun\Desktop\Data Analysis\CF\Data Immersion\Achievement 6\Ex 6.1\Gun Violence Data\cleaned_gun_violence_data.csv')